airflow/models/dag.py

```diff
@@ -2625,6 +2625,18 @@ def validate_schedule_and_params(self):
                     "DAG Schedule must be None, if there are any required params without default values"
                 )
 
+    @provide_session
+    def validate_task_pools(self, session=NEW_SESSION):
+        """Validates and raise exception if any task in a dag is using a non-existent pool"""
+        from airflow.models.pool import Pool
+
+        pools = [p.pool for p in Pool.get_pools(session)]
```

Comment:
Do you think it would be more performant to use a set instead of a list here since we're doing potentially a lot of `task not in pools`? 

The list is pretty small so I'm not sure how significant the difference would be.

Reply:
Makes sense. Have used set to also get all the wrong pools